src/evaluators/gather/PHAL_GatherScalarNodalParameter_Def.hpp

```diff
@@ -211,58 +213,76 @@ evaluateFields(typename Traits::EvalData workset)
     const int neq = dof_mgr->getNumFields();
     const int num_deriv = this->numNodes;
     bool trans = workset.transpose_dist_param_deriv;
-    const auto elem_dof_lids = dof_mgr->elem_dof_lids().host();
-    for (std::size_t cell=0; cell<workset.numCells; ++cell) {
+    const auto elem_dof_lids = dof_mgr->elem_dof_lids().dev();
+
+    Kokkos::parallel_for(this->getName(),RangePolicy(0,workset.numCells),
+                        KOKKOS_CLASS_LAMBDA(const int& cell) {
       const auto elem_LID = elem_lids(cell);
       const auto p_dof_lids = Kokkos::subview(p_elem_dof_lids,elem_LID,ALL);
       for (int node=0; node<num_deriv; ++node) {
         const LO lid = p_dof_lids(node);
 
         // Initialize Fad type for parameter value
-        const auto p_val = lid>=0 ? p_data[lid] : 0;
+        const auto p_val = lid>=0 ? p_data(lid) : 0;
         ParamScalarT v(num_deriv, node, p_val);
```

Comment:
Does this work on device, and is it performant? With certain fads, doesn't it allocate memory on device at every call?